### PR TITLE
Fixed Overfitting with Dropout Rate, Weight Decay, and on the fly data augmentation.

### PR DESCRIPTION
**Pull request by sridhs21**
# Add Regularization and Data Augmentation

This PR adds regularization techniques and on-the-fly data augmentation to improve model training.

### New Command-Line Arguments
* `--dropoutRate` (default: 0.3) - Control dropout strength
* `--weightDecay` (default: 5e-4) - Control L2 regularization
* `--seed` (default: None) - Set random seed for reproducibility
* `--require-gpu` - Exit if GPU not available

### Dropout Improvements
* `UNet` now accepts `dropout_rate` parameter
* Added dropout layers after each decoder block (`dec1_dropout`, `dec2_dropout`, `dec3_dropout`, `dec4_dropout`)
* Added dropout after bottleneck (`bottleneck_dropout`)

### Weight Decay
* Changed from hardcoded `weight_decay=1e-5` to configurable `weight_decay=args.weightDecay`
* Logged in optimizer initialization output

### On-the-Fly Data Augmentation
* Added `augment` parameter to `XPointPatchDataset`
* Added `seed` parameter to `XPointPatchDataset` for reproducible augmentation
* New `_apply_augmentation()` method with:
  - Random rotation (90°, 180°, 270°) - 75% probability
  - Random horizontal flip - 50% probability
  - Random vertical flip - 50% probability
  - Gaussian noise injection - 30% probability
  - Brightness/contrast adjustment - 30% probability
  - Random cutout/erasing - 20% probability
* Training uses `augment=True`, validation uses `augment=False`
* Removed static augmentation: `XPointDataset(..., rotateAndReflect=False)`

### Reproducibility
* New `set_seed()` function sets seeds for Python, NumPy, and PyTorch
* Configures PyTorch deterministic mode when seed is provided
* Seeds propagated to patch datasets

### GPU Checking
* `--require-gpu` flag exits with error if CUDA unavailable
* Provides diagnostic messages for troubleshooting

### CODE REVIEW

File: XPointMLTest.py [hunk 1]
```diff
@@ -381,7 +503,7 @@ class UNet(nn.Module):
     """
     Improved U-Net with residual blocks and better normalization
     """
-    def __init__(self, input_channels=4, base_channels=32):
+    def __init__(self, input_channels=4, base_channels=32, dropout_rate=0.2):
```
Comment (cwsmith) [inline XPointMLTest.py]: If there are no uses of this function that use the defaults it may be best to just make them required args.

**Review by cwsmith — CHANGES_REQUESTED** [review]
There is one small comment below.  In addition to that, we discussed offline about changing the default checkpoint rate to every 100 epochs.  Each checkpoint was about 100MB which could quickly fill up disk space for someone who doesn't expect it.

**Comment by cwsmith** [comment]
@sridhs21 would you also please change the default checkpoint interval to 100 epochs?

**Review by cwsmith — APPROVED** [review]
Thank you.
